src/utils/fees.spec.ts

```diff
@@ -0,0 +1,100 @@
+            },
+        };
+
+        const storageStats = [{
+            lastPaid: 1696792239, duePayment: null,
+            used: { bits: 6888, cells: 14, publicCells: 0 }
+        }]
+
+        const gasUsageByOutMsgs: { [key: number]: number } = { 1: 3308, 2: 3950, 3: 4592, 4: 5234 };
+
+        const contract = WalletContractV4.create({ workchain: 0, publicKey: Buffer.from('MUP3GpbKCQu64L4PIU0QprZxmSUygHcaYKuo2tZYA1c=', 'base64') });
+
+        const body = comment('Test message fees estimation');
+        const testAddress = Address.parse('EQCD39VS5jcptHL8vMjEXrzGaRcCVYto7HUn4bpAOg8xqB2N');
+
+        // Create transfer
+        let intMessage = internal({
+            to: testAddress,
+            value: 1400000000n,
+            bounce: true,
+            body,
+        });
+
+        let transfer = contract.createTransfer({
+            seqno: 14,
+            secretKey: Buffer.alloc(64),
+            sendMode: SendMode.IGNORE_ERRORS | SendMode.PAY_GAS_SEPARATELY,
+            messages: [intMessage]
+        });
+
+        const externalMessage = external({
+            to: contract.address,
+            body: transfer,
+            init: null
+        });
+
+        let inMsg = new Cell().asBuilder();
+        storeMessage(externalMessage)(inMsg);
+
+        let outMsg = new Cell().asBuilder();
+        storeMessageRelaxed(intMessage)(outMsg);
+
+        // Storage fees
+        let storageFees = BigInt(0);
+        for (let storageStat of storageStats) {
+            if (storageStat) {
+                const computed = computeStorageFees({
+                    lastPaid: storageStat.lastPaid,
+                    masterchain: false,
+                    now: Math.floor(Date.now() / 1000),
+                    special: false,
+                    storagePrices: config.storage,
+                    storageStat: {
+                        bits: storageStat.used.bits,
+                        cells: storageStat.used.cells,
+                        publicCells: storageStat.used.publicCells
+                    }
+                });
+                storageFees = storageFees + computed;
+            }
+        }
+
+        // Calculate import fees
+        let importFees = computeExternalMessageFees(config.workchain.message as any, inMsg.endCell());
+
+        // Any transaction use this amount of gas
+        const gasUsed = gasUsageByOutMsgs[1];
+        let gasFees = computeGasPrices(
+            BigInt(gasUsed),
+            { flatLimit: config.workchain.gas.flatLimit, flatPrice: config.workchain.gas.flatGasPrice, price: config.workchain.gas.price }
+        );
+
+
+        // Total
+        let total = BigInt(0);
+        total += storageFees;
+        total += importFees;
+        total += gasFees;
+
+        // Forward fees
+        let fwdFees = computeMessageForwardFees(config.workchain.message as any, outMsg.endCell());
+        total += fwdFees.fees;
+
+        console.log('Total fees:', fromNano(total));
```

Comment:
Where is comparisons?

Reply:
I've updated fees.spec.ts to improve fee calculation tests